src/functions.jl

```diff
@@ -53,82 +53,82 @@ function open_png(filename::String)
 end
 
 function create_read_struct()
-    png_ptr = ccall((:png_create_read_struct, libpng), Ptr{Void}, (Cstring, Ptr{Void}, Ptr{Void}, Ptr{Void}),
+    png_ptr = ccall((:png_create_read_struct, libpng), Ptr{Nothing}, (Cstring, Ptr{Nothing}, Ptr{Nothing}, Ptr{Nothing}),
                     PNG_LIBPNG_VER_STRING, C_NULL, png_error_fn, png_warn_fn)
     png_ptr == C_NULL && error("Failed to create png read struct")
     return png_ptr
 end
 
 function create_info_struct(png_ptr)
-    info_ptr = ccall((:png_create_info_struct, libpng), Ptr{Void}, (Ptr{Void},), png_ptr)
+    info_ptr = ccall((:png_create_info_struct, libpng), Ptr{Nothing}, (Ptr{Nothing},), png_ptr)
     info_ptr == C_NULL && error("Failed to create png info struct")
     return info_ptr
 end
 
-function png_init_io(png_ptr::Ptr{Void}, fp::Ptr{Void})
-    ccall((:png_init_io, libpng), Void, (Ptr{Void}, Ptr{Void}), png_ptr, fp)
+function png_init_io(png_ptr::Ptr{Nothing}, fp::Ptr{Nothing})
+    ccall((:png_init_io, libpng), Nothing, (Ptr{Nothing}, Ptr{Nothing}), png_ptr, fp)
 end
 
-function png_set_sig_bytes(png_ptr::Ptr{Void})
-    ccall((:png_set_sig_bytes, libpng), Void, (Ptr{Void}, Cint), png_ptr, PNG_BYTES_TO_CHECK)
+function png_set_sig_bytes(png_ptr::Ptr{Nothing})
+    ccall((:png_set_sig_bytes, libpng), Nothing, (Ptr{Nothing}, Cint), png_ptr, PNG_BYTES_TO_CHECK)
 end
 
-function png_read_png(png_ptr::Ptr{Void}, info_ptr::Ptr{Void}, transforms::Int)
-    ccall((:png_read_png, libpng), Void, (Ptr{Void}, Ptr{Void}, Cint, Ptr{Void}), png_ptr, info_ptr, transforms, C_NULL)
+function png_read_png(png_ptr::Ptr{Nothing}, info_ptr::Ptr{Nothing}, transforms::Int)
+    ccall((:png_read_png, libpng), Nothing, (Ptr{Nothing}, Ptr{Nothing}, Cint, Ptr{Nothing}), png_ptr, info_ptr, transforms, C_NULL)
 end
 
-function png_destroy_read_struct(png_ptr::Ptr{Void}, info_ptr::Ptr{Void})
-    png_ptr_ptr = Ref{Ptr{Void}}(png_ptr)
-    info_ptr_ptr = Ref{Ptr{Void}}(info_ptr)
-    ccall((:png_destroy_read_struct, libpng), Void, (Ref{Ptr{Void}}, Ref{Ptr{Void}}, Ptr{Ptr{Void}}), png_ptr_ptr, info_ptr_ptr, C_NULL)
+function png_destroy_read_struct(png_ptr::Ptr{Nothing}, info_ptr::Ptr{Nothing})
+    png_ptr_ptr = Ref{Ptr{Nothing}}(png_ptr)
+    info_ptr_ptr = Ref{Ptr{Nothing}}(info_ptr)
+    ccall((:png_destroy_read_struct, libpng), Nothing, (Ref{Ptr{Nothing}}, Ref{Ptr{Nothing}}, Ptr{Ptr{Nothing}}), png_ptr_ptr, info_ptr_ptr, C_NULL)
 end
 
-function close_png(fp::Ptr{Void})
-    ccall((:fclose, "libc"), Cint, (Ptr{Void},), fp)
+function close_png(fp::Ptr{Nothing})
+    ccall((:fclose, "/lib/x86_64-linux-gnu/libc.so.6"), Cint, (Ptr{Nothing},), fp)
```

Comment:
```suggestion
    ccall(:fclose, Cint, (Ptr{Nothing},), fp)
```

Hard coding this path seems like it will lead to trouble :-) (certainly it won't work on any non 64-bit or non-linux system). libc is already loaded into the julia process so you should be able to get away without mentioning the library at all.